components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2738,4 +2760,84 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+        if (typeof usageLimit !== "number" || usageLimit < 0) {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Unexpected usageLimit value: ${usageLimit}`);
+        }
+        const user = this.checkAndBlockUser("adminSetUsageLimit");
+        await this.guardAdminAccess("adminSetUsageLimit", { id: user.id }, Permission.ADMIN_USERS);
+
+        const response = await this.usageService.getCostCenter({ attributionId });
+
+        // backward compatibility for cost centers that were created before introduction of BillingStrategy
+        if (!response.costCenter) {
+            throw new ResponseError(ErrorCodes.NOT_FOUND, `Coudln't find cost center with id ${attributionId}`);
+        }
+        const stripeSubscriptionId = await this.findStripeSubscriptionId(ctx, attributionId);
+        if (stripeSubscriptionId != undefined) {
+            response.costCenter.billingStrategy = CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE;
+        }
+
+        await this.usageService.setCostCenter({
+            costCenter: {
+                attributionId,
+                spendingLimit: usageLimit,
+                billingStrategy: response.costCenter.billingStrategy,
+            },
+        });
+
+        this.messageBus.notifyOnSubscriptionUpdate(ctx, attrId).catch();
+    }
+
+    async adminListUsage(ctx: TraceContext, req: ListUsageRequest): Promise<ListUsageResponse> {
+        traceAPIParams(ctx, { req });
+        const user = this.checkAndBlockUser("adminListUsage");
+        await this.guardAdminAccess("adminListUsage", { id: user.id }, Permission.ADMIN_USERS);
+        return this.internalListUsage(ctx, req);
+    }
+
+    async adminGetUsageBalance(ctx: TraceContext, attributionId: string): Promise<number> {
+        traceAPIParams(ctx, { attributionId });
+        const user = this.checkAndBlockUser("adminGetUsageBalance");
+        await this.guardAdminAccess("adminGetUsageBalance", { id: user.id }, Permission.ADMIN_USERS);
+        const result = await this.usageService.getBalance({ attributionId });
+        return result.credits;
+    }
+
+    async adminAddUsageCreditNote(
+        ctx: TraceContext,
+        attributionId: string,
+        credits: number,
+        note: string,
+    ): Promise<void> {
+        traceAPIParams(ctx, { attributionId, credits, note });
+        const user = this.checkAndBlockUser("adminAddUsageCreditNote");
+        await this.guardAdminAccess("adminAddUsageCreditNote", { id: user.id }, Permission.ADMIN_USERS);
+        await this.usageService.addUsageCreditNote({
+            attributionId,
+            credits,
+            note,
+            userId: user.id,
+        });
```

Comment:
Would it be possible to force a human description to be included here? It's the only way can actually track WHY it was done, as opposed to that it was done.

Reply:
That's what the 'note' variable contains